### DISCUSSION TITLE
Standalone dependency injection package

### DISCUSSION THREAD

**Discussion by dennis-rall** [fastapi/fastapi#9819] [Questions] [answered]
Have you ever thought of building a standalone python package out of the dependency mechanism?

- then you could use dependency injection in projects that don't rely on FastAPI (e.g., Typer)
- it would provide a better separation of concerns as utilizing dependency injection in business services would not require a coupling to an API framework

What do you think about this? And how complex would this be?

**Comment by jgould22** [comment, marked as answer]
You can see someone experimenting with that idea here https://github.com/adriangb/xpresso
